nixos/modules/system/etc/build-composefs-dump.py

```diff
@@ -83,8 +83,12 @@ def write_line(self) -> str:
         return " ".join(line_list)
 
 
-def eprint(*args, **kwargs) -> None:
-    print(args, **kwargs, file=sys.stderr)
+def eprint(*args: Any, **kwargs: Any) -> None:
+    print(*args, **kwargs, file=sys.stderr)
+
+
+def normalize_path(path: str) -> str:
+    return str("/" + os.path.normpath(path).lstrip("/"))
```

Comment:
Note that normpath has this note [here](https://docs.python.org/3/library/os.path.html#os.path.normpath)

> On POSIX systems, in accordance with [IEEE Std 1003.1 2013 Edition; 4.13 Pathname Resolution](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap04.html#tag_04_13), if a pathname begins with exactly two slashes, the first component following the leading characters may be interpreted in an implementation-defined manner, although more than two leading characters shall be treated as a single character.

And indeed, `os.path.normpath("//foo")` returns `'//foo'`. `lstrip('/')` works regardless to remove the two slashes but I don't know what exactly the implementation defined behavior is here, it might not for a different input. To avoid this, maybe ensure that there will never be exactly two slashes:

```suggestion
    return str("///" + os.path.normpath(path).lstrip("/"))
```

No idea if this is necessary though.

Reply:
Not sure I understand why you want triple `/` here? I think the code should work as it is..

Reply:
> maybe ensure that there will never be exactly two slashes

I'm doing something even better: I ensure that the path I have just normalized doesn't haven *any* leading slash and then only add the one leading slash I need.

Reply:
Oh, I'm an idiot. Of course I mean this:

```suggestion
    return str("/" + os.path.normpath('///' + path).lstrip("/"))
```

> Not sure I understand why you want triple / here? I think the code should work as it is..

Because of the implementation defined behavior for paths beginning with //, as I said.

Reply:
ah, now it makes sense. Should not do any harm to use triple slashes here, but I think we should add a reference to that IEEE Standard in the code, since it's otherwise not obvious at all, why we do that